APE_specutil.rst

```diff
@@ -0,0 +1,481 @@
+
+Dimensionality
+^^^^^^^^^^^^^^
+
+``flux`` and ``uncertainty`` may be multidimensional as long as the last
+dimension matches the shape of ``spectral_axis`` This is meant for fast
+operations on collections of spectra that share the same spectral_axis.  While
+it may seem to conflict with the “1D” in the class name, this name scheme is
+meant to communicate the presence of a single common spectral axis.  For more
+on this, see the section on data cubes below.
+
+Metadata
+^^^^^^^^
+
+Note that the ``meta`` attribute is meant for metadata, e.g., information from
+a FITS header.  A ``Spectrum1D`` object may include additional attributes
+beyond the metadata. In many cases this is best implemented via *subclasses* of
+``Spectrum1D``, but these attributes can also be applied directly by analysis
+steps discussed in the “Analysis tools to be included in specutils” section
+below.  For example, a background subtraction tool would add an attribute
+``background`` to the result spectrum, which is itself a ``Spectrum1D`` with
+the same shape as this one which contains the subtracted background.
+
+Creating Spectrum1D objects
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+While the above properties specify the interface for accessing a `Spectrum1D`,
+these are *not* all required for initialization. At the minimum, the user will
+have to provide a ``flux`` (which can be a Quantity or an array) and the
+``spectral_axis``. The Spectrum1D’s spectral axis can be instantiated with
+either a Quantity object, array, or a pixel-to-spectral transformation function
+(the `wcs` object).
+
+In addition to this information, `specutils` will provide a method for reading
+and writing formats regularly used for spectroscopy.  Common formats include
+IRAF, SDSS, MIDAS, FITS tables, and ascii tables.  This I/O Reading and writing
+a `Spectrum1D` object will use the astropy I/O registry.  A
+`Spectrum1D.create_from_array` method should be available to directly create a
+`Spectrum1D` object without the user having to interface with the ``wcs``
+property.
+
+Arithmetic Operations
+^^^^^^^^^^^^^^^^^^^^^
+
+In addition, methods should exist for arithmetic on `Spectrum1D` objects.
+Objects with different `spectral_axis`’s will raise an error if attempts are
+made to apply arithmetic to them.  Arithmetic on non-matching spectral axes
+therefore requires explicit interpolation so as not to make assumptions about
+the “best” interpolation algorithm for a given use case. `Spectrum1D` will
+contain interpolation methods, either onto different wavelengths (with
+flux-conserving at least an option) or over bad data points.  It will include
+methods for transforming the spectra to different unit systems (e.g.,
+converting between different velocity frames, frequency vs wavelength, flux
+transformations, air to vacuum). Note that in general these operations yield
+*new* `Spectrum1D` objects rather than changing anything in place.
+
```

Comment:
How would the wcs information be propagated in (1) the case of one of the spectra in the operation having a wcs object, and (2) the case where both have wcs information? Should only the wcs of the object whose spectral axis is not interpolated be stored, and the other's stored in the `meta` attribute?

Reply:
I think the WCS would have the `spectral_frame` property and so on these transformations that will be updated in the new `Spectrum1D` would update the WCS property.   As mentioned in arithmetic, operations on incompatible `Spectrum1D` objects would raise an error.

Reply:
Yeah, I think the *general* solution to this is very hard, and it's safe to say, at least in an initial implementation, that you'd basically always "downgrade" to just a sampled `spectral_axis`. Anything more clever might be situation-specific and left to the user.